### ISSUE TITLE
&broadcast not working on iOS (so use PiP instead?)

### ISSUE THREAD

**Issue by steveseguin** [steveseguin/obsninja#570]
Posted byu/mikemaan
5 minutes ago
&broadcast not working on iOS
Bug Report
Whenever I use the &broadcast parameter on iOS the camera doesn't stream. When I remove that parameter everything works as expected. Is this a bug or could it be something else?

2019 iPad Pro iOS 14.0.1

https://www.reddit.com/r/OBSNinja/comments/k6usd6/broadcast_not_working_on_ios/

**Comment by steveseguin** [comment]
There does seem to be an issue on iOS devices where if the camera preview isn't visible on screen, then it won't send any video out -- it will just push a black video frame.  

I pushed a hotfix to production that has the &broadcast flag not enable the "no preview" mode for iOS devices.  This should allow the &broadcast flag to work in the mean time, even if there will be up to two videos on screen at a time.

The long term goal I had for the broadcast flag was to have the self-preview be a small floating window anyways, so I think this isn't a huge problem.  I'd love to hear feedback on how you'd ideally like the self-preview to be handled though.

**Comment by chrismarquardt** [comment]
I tried to reproduce the issue but tried so after you had pushed the hotfix and thought I had gone crazy for a minute. Thanks for this workaround!

**Comment by steveseguin** [comment]
I pushed the hotfix last night, but my iPhone was running a beta version of iOS and it was causing it's own serious problems when I was trying to confirm that was the only issue. Didn't want to assume I fixed it.  Finished updating the phone this morning to new beta build though and the crashes I was having are gone. 

In the obs.ninja/alpha build, I fixed an issue with the director's camera not preferring 16:9 as an aspect ratio; on production it's set to just auto-mode. I also noticed some issues where the new camera re-connection logic on alpha was being triggered when it shouldn't be with iOS devices -- iOS devices are mighty strange things.  For the time being, I've had to disable that logic for iOS also, but will add it back in once I find a solution. Adding/removing new devices shouldn't be an issue with iOS devices really anyways.

I'll probably make the preview camera on iOS a smaller floating preview window longer term, unless there is a better idea given.
